extensions/resource-deployment/src/helpers/optionSources.ts

```diff
@@ -11,74 +11,112 @@ import { apiService } from '../services/apiService';
 import { throwUnless } from '../utils';
 import { CacheManager } from './cacheManager';
 
-export enum OptionsSourceType {
-	ArcControllersOptionsSource = 'ArcControllersOptionsSource'
+export const enum OptionsSourceType {
+	ArcControllersOptionsSource = 'ArcControllersOptionsSource',
```

Comment:
I've created an issue to not have these be hardcoded in resource deployment. That isn't maintainable long term to have to have this extension know about all the possible options

https://github.com/microsoft/azuredatastudio/issues/12682

Please work on that task when you finish up with your current stuff - it would be nice to get this done before we add too many other option types. 

(this is ok for now in this PR but  I'd like this done before we have to add another type)